src/NuGet.Core/NuGet.Commands/RestoreCommand/DependencyGraphResolver.cs

```diff
@@ -88,6 +91,8 @@ public DependencyGraphResolver(RestoreCollectorLogger logger, RestoreRequest res
                     hasInstallBeenCalledAlready = true;
                 }
 
+                TargetFrameworkInformation? projectTargetFramework = _request.Project.TargetFrameworks.FirstOrDefault(i => NuGetFramework.Comparer.Equals(i.FrameworkName, pair.Framework));
```

Comment:
There's already extension methods for this: https://github.com/NuGet/NuGet.Client/blob/f06d028edc7851dff7014fdca5e360667d72e8f7/src/NuGet.Core/NuGet.ProjectModel/PackageSpecExtensions.cs#L15-L25.

Reply:
I'll do this in the refactor and tracking it in https://github.com/NuGet/Client.Engineering/issues/2959